server/index.ts

```diff
@@ -8,7 +9,10 @@ import closeWithGrace from 'close-with-grace'
 import { createRequestHandler } from '@remix-run/express'
 import { broadcastDevReady } from '@remix-run/node'
 
-const BUILD_DIR = path.join(process.cwd(), 'build')
+const pathParts = [process.cwd(), 'build']
+if (process.env.NODE_ENV === 'production') pathParts.push('index.js')
+const BUILD_DIR = path.join(...pathParts)
+const BUILD_DIR_FILE_URL = pathToFileURL(BUILD_DIR).href
```

Comment:
Does pushing `index.js` on here not work in non-production environments? Based on what I'm reading below, I'm pretty sure we can rework a few things so we can just do:

```suggestion
const BUILD_DIR = path.join(process.cwd(), 'build')
const BUILD_DIR_FILE = path.join(BUILD_DIR, 'index.js')
const BUILD_DIR_FILE_URL = pathToFileURL(BUILD_DIR).href
```

And then make a few adjustments below.

In fact, I wonder whether `BUILD_DIR_FILE_URL` is necessary, couldn't we just `await import(BUILD_DIR_FILE)`?

Reply:
We need to use `BUILD_DIR_FILE_URL` on Windows, see this error message

```shell
Error: Only URLs with a scheme in: file and data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at new NodeError (node:internal/errors:399:5)
    at throwIfUnsupportedURLScheme (node:internal/modules/esm/resolve:1059:11)
    at defaultResolve (node:internal/modules/esm/resolve:1135:3)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at u (file:///C:/projects/kcd-apps/epic-stack/node_modules/@esbuild-kit/esm-loader/dist/index.js:1:2406)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at nextResolve (node:internal/modules/esm/loader:163:22)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:24)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:7) {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
}
```

we can use `const BUILD_DIR = '../build/index.js'`